packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -406,6 +406,12 @@ export class DocumentDeltaConnection
                 let isWebSocketTransportError = false;
                 try {
                     const description = error?.description;
+                    const responseText = error?.context?.responseText;
+
+                    if (responseText?.includes("Error: self signed certificate")) {
+                        failAndCloseSocket(this.createErrorObject("connect_error", error, false));
+                    }
```

Comment:
A couple of things here:

- I think we can handle the error a bit earlier by checking what `error?.description` says. Running some checks, it seems we pass through here a few times before `responseText` indicates anything about the self-signed certificate, but `description.message` already makes it clear that's the issue.
- We should `return;` immediately after the call to `failAndCloseSocket()`. Based on how it's used in other places in this file (and the fact that it doesn't throw, it just "sets up" a promise to be rejected), it doesn't prevent further code from executing, so right now we're hitting this call and then still flowing through the rest of the method (even calling this function again before exiting the handler).
- I'm getting weird behavior where if I run with `--inspect-brk` and put breakpoints around this, then when "letting the breakpoint go", test execution terminates immediately but the error is not printed to console. And if I don't specify `--inspect-brk`, the test does not fail, it eventually hits the mocha timeout.

Let's investigate that a bit more before merging this change. Depending on how that investigation goes, maybe we split the piece for the websocket connection into its own PR so we can at least merge the other piece.

Reply:
I made the change by adding an if-statement for `if (description.message?.includes("self signed certificate"))` which will run checks on `description` whether it has `self signed certificate` error or not.